teuthology/provision/fog.py

```diff
@@ -278,7 +278,11 @@ def _wait_for_ready(self):
                     NoValidConnectionsError,
                     MaxWhileTries,
                     EOFError,
-                ):
+                ) as e:
+                    # log this, because otherwise lots of failures just keep retrying without
+                    # any notification (like, say, a mismatched host key in ~/.ssh/known_hosts, or
+                    # something)
+                    log.exception(e)
```

Comment:
`log.exception` logs the traceback automatically - the arg should be a string like "exception encountered while __"